controller/api/public/stat_summary.go

```diff
@@ -197,13 +195,12 @@ func processRequests(vec model.Vector, labelSelector string) map[string]*pb.Basi
 	return result
 }
 
-func (s *grpcServer) getDeployment(namespace string, name string) ([]appsv1.Deployment, map[string]*meshedCount, error) {
+func (s *grpcServer) getDeployment(namespace string, name string) ([]*appsv1.Deployment, map[string]*meshedCount, error) {
```

Comment:
The listers that we're introducing in this branch return pointers to k8s objects, so I've updated these method signatures to match. That allows us to avoid manipulating the response that come back from the cache.